paper/paper.md

```diff
@@ -0,0 +1,248 @@
+modifying the indicators, change metrics and window type of the code shown above:
+
+```julia
+# Here the data should not be detrended
+time, data = load_data()
+
+indicators = (nothing, nothing)
+change_metrics = (difference_of_mean(), difference_of_max())
+config = SlidingWindowConfig(indicators, change_metrics;
+   width_cha = 50, whichtime = midpoint)
+```
+
+This example showcases that steps of the analysis pipeline can be skipped altogether, as
+done here for the computation of indicators, since the change
+metric of the original time series is already sufficient to detect a transition by comparing
+the difference in mean and maximum values between the two halves of the sliding window.
+Furthermore, the user can define their own `IndicatorChangesConfig` - for instance instead of
+`SlidingWindowConfig` - and are therefore capable of adding new ways of computing the
+relevant metrics. Finally, different ways of testing for significance can be interchangeably
+used, therefore expanding the versatility of the software beyond the surrogate analysis.
+
+## Extensibility
+
+An important aspect of the modularity mentioned above, is that self-written functions can be
+passed as indicators or change metrics to the analysis pipeline. Thus, researchers can easily
+test new methods without any programming overhead, nor modification of the source code. To
+illustrate this, the code shown above can include the skewness as indicator of the
+transition by modifying a few lines:
+
+```julia
+skewness(x::Vector) = mean( (x .- mean(x))^3 ) / mean( (x .- mean(x))^2 )^1.5
+indicators = (var, ar1_whitenoise, skewness)
+change_metrics = (RidgeRegressionSlope(), RidgeRegressionSlope(), RidgeRegressionSlope())
+```
+
+There is no complexity restriction on the self-programmed functions, as long as they comply
+with the structure of taking a vector as input and returning a scalar as output. Finally,
+users can modify the fully-open source code, written in native Julia and thus offering a good
+legibility compared to other high-performance languages.
+
+## Integration
+
+TransitionsInTimeseries.jl is designed to be well integrated into the Julia ecosystem.
+Functions can be imported from other packages and subsequently passed as indicators or
+change metrics. For instance, the skewness implemented above can be loaded from another julia
+package instead of being programmed by the user:
+
+```julia
+using StatsBase
+indicators = (var, ar1_whitenoise, StatsBase.skewness)
+```
+
+TransitionsInTimeseries.jl therefore offers an extremely wide and
+potentially unlimited library of indicators. Furthermore, TimeseriesSurrogates.jl
+[@haaga_timeseriessurrogatesjl_2022] is used to create surrogates of the time series,
+thus offering optimized routines with numerous surrogate types. Finally,
+`plot_changes_significance` showcased in [Fig. 1](@figure1) relies on Makie.jl, which allows
+the user to customize the figures according to their needs.
```

Comment:
I think all of this section is too large and should be significantly decreased in size. Above comment says how to mention impoting functions from other packages. And we definitely do not need a code example for using functions from other packages.

Makie.jl is **not an integration**. Iti s a dependency. Its not like the user can interface with the plotting code in any meaningful way. 

You can mention the surrogate integration though.

Reply:
p.s.: We can add the skewness example to the main tutorial to show defining user defined functions or importing new ones.